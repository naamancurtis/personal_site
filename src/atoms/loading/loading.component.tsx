import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { LoadingWrapper, LoadingText } from './loading.styles';

const Loading = () => {
  return (
    <LoadingWrapper>
      <FontAwesomeIcon icon={['fas', 'spinner']} className="fa-pulse" />
      <LoadingText>Loading...</LoadingText>
    </LoadingWrapper>
  );
};

export default Loading;
