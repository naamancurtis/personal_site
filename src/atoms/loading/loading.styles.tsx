import styled from 'styled-components';

export const LoadingWrapper = styled.div`
  width: 90vw;
  height: 90vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 10rem;

  background-color: ${({ theme }) => theme.body};
  color: ${({ theme }) => theme.text};
`;

export const LoadingText = styled.h2`
  padding-top: 1.5rem;
  color: ${({ theme }) => theme.text};
  font-size: 30px;
`;
