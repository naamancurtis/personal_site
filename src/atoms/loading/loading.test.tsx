import React from 'react';
import { shallow } from 'enzyme';
import { Loading } from './loading.component';

describe('<Loading />', () => {
  it('expect to render component', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
