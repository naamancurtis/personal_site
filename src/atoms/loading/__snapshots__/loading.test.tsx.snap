// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> expect to render component 1`] = `
<styled.div>
  <FontAwesomeIcon
    border={false}
    className="fa-pulse"
    fixedWidth={false}
    flip={null}
    icon={
      Array [
        "fas",
        "spinner",
      ]
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
  <styled.h2>
    Loading...
  </styled.h2>
</styled.div>
`;
